lib/main-window.js

```diff
@@ -266,78 +257,99 @@ module.exports = function (config) {
 
   // scoped
 
-  function subMenu (label, items) {
-    return function () {
-      return {
-        label,
-        submenu: items.map(item => {
-          return {
-            label: item[0],
-            click () {
-              navigate(item[1])
-            }
-          }
-        })
-      }
-    }
-  }
-
   function getSubscribedChannelMenu () {
     const channels = Array.from(subscribedChannels()).sort(localeCompare)
 
     if (channels.length) {
       return {
+        type: 'submenu',
         label: i18n('Channels'),
         submenu: [
           {
+            type: 'normal',
             label: i18n('Browse Recently Active'),
-            click () {
-              navigate('/channels')
-            }
+            target: '/channels'
           },
           { type: 'separator' }
-        ].concat(channels.map(channel => {
-          return {
-            label: `#${channel}`,
-            click () {
-              navigate(`#${channel}`)
-            }
-          }
-        }))
+        ].concat(channels.map(channel => ({
+          type: 'normal',
+          label: `#${channel}`,
+          target: `#${channel}`,
+        })))
       }
     } else {
       return {
+        type: 'normal',
         label: i18n('Browse Channels'),
-        click () {
-          navigate('/channels')
-        }
+        target: '/channels',
       }
     }
   }
 
-  function dropTab (title, items) {
+  function buildDropdownMenuItems() {
```

Comment:
In hindsight, this could be done in the main process.
But even then, I'd probably build it like this for legibility and keep the `buildMenu()` around to add the `navigate-to` event handlers afterwards. So... dunno, feels okay to have the description of the page structure in the renderer process, that's where the rest of it lives too, right?